src/strategies/optimism/velodrome/strategy-velo-base.sol

```diff
@@ -0,0 +1,165 @@
+
+        gauge = _gauge;
+        token0 = IUniswapV2Pair(_lp).token0();
+        token1 = IUniswapV2Pair(_lp).token1();
+
+        IERC20(native).approve(solidRouter, type(uint256).max);
+        IERC20(token0).approve(solidRouter, type(uint256).max);
+        IERC20(token1).approve(solidRouter, type(uint256).max);
+        IERC20(velo).approve(solidRouter, type(uint256).max);
+        IERC20(want).approve(gauge, type(uint256).max);
+    }
+
+    function balanceOfPool() public view override returns (uint256) {
+        return ISolidlyGauge(gauge).balanceOf(address(this));
+    }
+
+    function getHarvestable() external view override returns (address[] memory, uint256[] memory) {
+        uint256[] memory pendingRewards = new uint256[](1);
+        address[] memory rewardTokens = new address[](1);
+        pendingRewards[0] = ISolidlyGauge(gauge).earned(velo, address(this));
+        rewardTokens[0] = velo;
+
+        return (rewardTokens, pendingRewards);
+    }
+
+    // **** Setters ****
+
+    function deposit() public override {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            ISolidlyGauge(gauge).deposit(_want, 0);
+        }
+    }
+
+    function _withdrawSome(uint256 _amount) internal override returns (uint256) {
+        ISolidlyGauge(gauge).withdraw(_amount);
+        return _amount;
+    }
+
+    // **** State Mutations ****
+
+    // Adds/updates a swap path from a token to native, normally used for adding/updating a reward path
+    function addToNativeRoute(ISolidlyRouter.route[] calldata path) external {
+        require(msg.sender == timelock, "!timelock");
+        _addToNativeRoute(path);
+    }
+
+    function _addToNativeRoute(ISolidlyRouter.route[] memory path) internal {
+        if (toNativeRoutes[path[0].from].length == 0) {
+            activeRewardsTokens.push(path[0].from);
+        }
+        for (uint256 i = 0; i < path.length; i++) {
+            toNativeRoutes[path[0].from].push(path[i]);
+        }
+        IERC20(path[0].from).approve(solidRouter, type(uint256).max);
+    }
+
+    function deactivateReward(address reward) external {
+        require(msg.sender == timelock, "!timelock");
+        for (uint256 i = 0; i < activeRewardsTokens.length; i++) {
+            if (activeRewardsTokens[i] == reward) {
+                activeRewardsTokens[i] = activeRewardsTokens[activeRewardsTokens.length - 1];
+                activeRewardsTokens.pop();
+            }
+        }
+    }
+
+    function _swapSolidlyWithRoute(ISolidlyRouter.route[] memory routes, uint256 _amount) internal {
+        require(routes[0].to != address(0));
+        ISolidlyRouter(solidRouter).swapExactTokensForTokens(_amount, 0, routes, address(this), block.timestamp + 60);
+    }
+
+    function harvest() public override {
+        // Collects rewards tokens
+        address[] memory _rewardsAddresses = new address[](1);
+        _rewardsAddresses[0] = velo;
```

Comment:
Why is this being re-assigned on every `harvest` call? It can be done in the constructor since it stays constant, and I don't see any use of `_rewardsAddresses` either, so is this an artifact that should be removed?